src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1763,6 +1710,234 @@ println outFile.text
+
+Task _B_1_ can use Task Synchonization API to wait until the apache server is advertised, it can also perform the necessary operations and advertise task _A_1_ that the operation is complete and the apache server can be terminated.
+
+As we can see, Task Synchronization API can be used for two-ways synchronization.
+
+==== API Usage
+
+*Task Synchronization API* is a *Java API* defined by the following https://www.activeeon.com/public_content/documentation/javadoc/dev/org/ow2/proactive/scheduler/synchronization/Synchronization.html[API documentation].
+
+The Synchronization API is connected to a centralized *Synchronization Service*. This service ensures that each request is executed atomically, and that changes are persisted on disk.
+
+The *Synchronization Service* works as a _Key-Value Store_ organized in *channels*. Each *channel* acts as a private _Key-Value Store_ or _Hash Map_.
+
+This means that each _put_ operation is identified by a 3-tuple: _<Channel_identifier, Key_binding, Value>_.
+
+The Synchronization API is available inside ProActive Tasks through the `synchronizationapi` <<_glossary_script_bindings,script binding>>.
+
+Users of the Synchronization API are responsible to create *channels* as fit their needs. For example, a channel which is used inside a specific job should be created inside some *initialization task* at the beginning of this job, and deleted in some *cleaning task* at the end.
+When channels are created, the user can decide if the channel should be persisted on disk or kept in memory only.
+
+The Synchronization API is similar to the https://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Java 8 Map interface], with the following differences:
+
+ * Each operation must take a _channel identifier_ parameter. Some methods like _createChannel_ allows to manipulate channels.
+
+ * The Synchronization API supports https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html[Java 8 lambdas]. All methods present in the https://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Java 8 Map interface] are implemented, but lambdas must be passed as string arguments defining http://groovy-lang.org/closures.html[Groovy Closures].
+
+ * The Synchronization Service guaranties that multiple operations, coming from different ProActive Tasks, will all be executed *atomically*.
+
+ * In addition to the methods present in the https://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Java 8 Map interface], Synchronization API provides *blocking methods* to wait for a specific *condition* on a channel.
+ This is mainly why this API is called *Synchronization*, it allows for different ProActive Tasks to be blocked until specific conditions are met.
+
+
+==== Example 1 : Simple Key/Value With Selection Script
+
+In this first example, we will show how one task can delay the execution of another task using the Synchronization API and a <<_selection,selection script>>.
```

Comment:
The user might ask himself why doing that in the selection script instead of within the task.